src/ucs/vfs/base/vfs_obj.h

```diff
@@ -87,11 +100,14 @@ void ucs_vfs_obj_add_dir(void *parent_obj, void *obj, const char *rel_path, ...)
  *
  * @param [in] obj      Pointer to the object. @a rel_path is relative to @a obj
  *                      directory.
+ * @param [in] arg      Pointer to the optional argument that is passed to the
+ *                      callback method.
  * @param [in] text_cb  Callback method that generates the content of the file.
  * @param [in] rel_path Format string which specifies relative path to the file.
  */
-void ucs_vfs_obj_add_ro_file(void *obj, ucs_vfs_file_show_cb_t text_cb,
-                             const char *rel_path, ...) UCS_F_PRINTF(3, 4);
+void ucs_vfs_obj_add_ro_file(void *obj, void *arg,
+                             ucs_vfs_file_show_cb_t text_cb,
+                             const char *rel_path, ...) UCS_F_PRINTF(4, 5);
```

Comment:
usually , arg comes after the callback
e.g
void ucs_vfs_obj_add_ro_file(void *obj, text_cb, arg, ...)